torch/_dynamo/eval_frame.py

```diff
@@ -364,7 +365,25 @@ def lookup_backend(compiler_fn):
     if isinstance(compiler_fn, str):
         from .optimizations import BACKENDS
 
-        compiler_fn = BACKENDS[compiler_fn]
+        backend_name = compiler_fn
+        if backend_name not in BACKENDS:
+            compiler_fn = None
+            group_name = "torch_dynamo_backends"
+            if sys.version_info < (3, 10):
+                backend_eps = entry_points()
+                eps = [ep for ep in backend_eps[group_name] if ep.name == backend_name]
+                if len(eps) > 0:
+                    compiler_fn = eps[0].load()
+            else:
+                backend_eps = entry_points(group=group_name)
+                if backend_name in backend_eps.names:
+                    compiler_fn = backend_eps[backend_name].load()
+            if compiler_fn is None:
+                raise RuntimeError("Cannot recognize backend: {}".format(backend_name))
+            BACKENDS[backend_name] = compiler_fn
+
+        compiler_fn = BACKENDS[backend_name]
```

Comment:
Please move all the logic to memoizable `find_backend_` function
```suggestion
        backend_name = compiler_fn
        compiler_fn = find_backend_(backend_name)
        if compiler_fn is None:
            raise RuntimeError("Cannot recognize backend: {}".format(backend_name))
```

Reply:
Hi @malfet, I memorize the found backend to `BACKENDS` at Line 383 (suggested by @dzhulgakov), thus we will not search it in ``entry_points`` in subsequent usage. I feel it slightly better to store both the predefined and external (i.e., through ``entry_points``) backends in the same place. 

Of course, we can also put all the logic (lookup `BACKENDS` and `entry_points`) in another function and cache it through `functools.lru_cache`. If you prefer this way, let me know and I will update.

Reply:
I have seen the refactored registration in the main branch, and seems it is better to use a new function to lookup the entry points backend. I will let you know when I finish rebase.